packages/lesswrong/server/scripts/resolveReviewMarkets.ts

```diff
@@ -0,0 +1,90 @@
+import { Posts } from "@/lib/collections/posts";
+import { manifoldAPIKeySetting } from "@/lib/instanceSettings";
+import { postGetMarketInfoFromManifold, LiteMarket } from "@/lib/collections/posts/annualReviewMarkets";
+import { Globals } from "@/lib/vulcan-lib";
+
+const manifoldAPIKey = manifoldAPIKeySetting.get()
+
+export const getMarketInfoFromManifold = async (marketId: string): Promise<LiteMarket | null > => {
+
+  const result = await fetch(`https://api.manifold.markets./v0/market/${marketId}`, {
+    method: "GET",
+    headers: {
+      "content-type": "application/json"
+    },
+  })
+  
+  if (!result.ok) {
+    //eslint-disable-next-line no-console
+    console.error(`HTTP error! status: ${result.status}`); 
+    return null
+  }
+
+  return result.json()
+}
+
+const resolvePosts = async (year: number, limit?: number) => {
+
+  const resolutionLimitOptions = limit ? {limit} : {}
+
+  const postsToResolve = await Posts.find({
+    manifoldReviewMarketId: {$exists: true},
+    createdAt: {$gte: new Date(`${year}-01-01`), $lte: new Date(`${year}-12-31`)}
+  }, resolutionLimitOptions).fetch()
+
+  const postsResolvingYes = await Posts.find({
+    createdAt: {$gte: new Date(`${year}-01-01`), $lte: new Date(`${year}-12-31`)},
+  }, {
+    limit: 50,
```

Comment:
I'm not sure what the exact resolution criteria were, but in the past we have in fact hand-curated some top-"50" posts which weren't strictly top-50 by review-vote-score as calculated by whatever algorithm we ran that year (and in some years we had less or more than 50 posts that ended up in the "winners" collection for that year)

Reply:
I decided to go with this criterion, tho I agree it's ambiguous